src/flag_gems/ops/cross_entropy_loss.py

```diff
@@ -488,47 +524,77 @@ def forward(ctx, inp, target, weight, reduction, ignore_index, label_smoothing):
         inp = inp.contiguous()
         tgt = target.contiguous()
         weight = weight.contiguous()
-        out = torch.zeros(shape, dtype=torch.float32, device=inp.device)
+        out = torch.empty(shape, dtype=torch.float32, device=inp.device)
         grid = lambda meta: (N, triton.cdiv(D, meta["BLOCK_D"]))
```

Comment:
I suggest reversing the order of the 2 dimensions, since when CTAs in a grid gets scheduled in an Fortran order. (Iterate grid.x, then grid.y, then grid.z).